modules/iam-group-with-policies/main.tf

```diff
@@ -51,7 +51,7 @@ resource "aws_iam_policy" "iam_self_management" {
 }
 
 resource "aws_iam_policy" "custom" {
-  count = length(var.custom_group_policies) > 0 ? length(var.custom_group_policies) : 0
+  count = length(var.custom_group_policies)
```

Comment:
looks good - might be worth considering a refactor to `for_each` with an array of maps. just food for thought

Reply:
ill go over he module and see where this refactor is needed. i wanted to make changes more incremental